portals/devportal/source/src/app/components/Apis/Details/SolaceApi/SolaceTopicsInfo.jsx

```diff
@@ -0,0 +1,580 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com). All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein is strictly forbidden, unless permitted by WSO2 in accordance with
+ * the WSO2 Commercial License available at http://wso2.com/licenses.
+ * For specific language governing the permissions and limitations under
+ * this license, please see the license as well as any agreement you’ve
+ * entered into with WSO2 governing the purchase of this software and any
+ * associated services.
+ */
+
+import Loading from 'AppComponents/Base/Loading/Loading';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useParams } from 'react-router-dom';
+import { makeStyles } from '@material-ui/core/styles';
+import API from 'AppData/api';
+import Typography from '@material-ui/core/Typography';
+import { FormattedMessage, useIntl } from 'react-intl';
+import {
+    Divider, Grid, MenuItem, TextField,
+} from '@material-ui/core';
+import { upperCaseString } from 'AppData/stringFormatter';
+import Box from '@material-ui/core/Box';
+import InputBase from '@material-ui/core/InputBase';
+import Avatar from '@material-ui/core/Avatar';
+import Tooltip from '@material-ui/core/Tooltip';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import Icon from '@material-ui/core/Icon';
+import Chip from '@material-ui/core/Chip';
+import ResourceNotFound from '../../../Base/Errors/ResourceNotFound';
+import solaceTopicStyles from './SolaceTopicStyles';
+
+
+function VerbElement(props) {
+    const {
+        verb,
+    } = props;
+
+    const useMenuStyles = makeStyles((theme) => {
+        const backgroundColor = theme.custom.resourceChipColors[verb.toLowerCase()];
+        return {
+            customButton: {
+                backgroundColor: '#ffffff',
+                borderColor: backgroundColor,
+                color: backgroundColor,
+                width: theme.spacing(2),
+            },
+        };
+    });
+    const classes = useMenuStyles();
+    return (
+        <Button disableFocusRipple variant='outlined' className={classes.customButton} size='small'>
+            {verb.toUpperCase()}
+        </Button>
+    );
+}
+
+function SolaceTopicsInfo() {
+    const classes = solaceTopicStyles();
+    const { apiUuid } = useParams();
+    const [application, setApplication] = useState(null);
```

Comment:
There are around 14 state variables here. It would be efficient to use a reducer if the state variables are intend manage a multiple parts of a single big data structure. For example if a UI is trying to manipulate a person object like the following.
```json
{
    "name": "chanaka",
    "age": "92",
    "school": "vanatamulla central",
    "spouse": { 
        "basic": "sandra", 
         "secret": "sara"
    }
}
```
Ref: https://reactjs.org/docs/hooks-reference.html#usereducer

This is something good to do.